fw/http.c

```diff
@@ -6782,6 +6782,18 @@ tfw_http_resp_process(TfwConn *conn, TfwStream *stream, struct sk_buff *skb)
 	 */
 	bad_req = hmresp->req;
 	tfw_http_popreq(hmresp, false);
+
+	/*
+	 * Special case: malformed response to a Health Monitor request.
+	 * There's no client involved, so just log-n-drop the response now 
+	 * without further processing.
+	 */
+	if(unlikely(test_bit(TFW_HTTP_B_HMONITOR, bad_req->flags))) {
+		T_WARN("Health Monitor response malformed");
+		tfw_http_resp_pair_free(bad_req);
+		return T_OK;
```

Comment:
I've doubts about the case when we got malformed response and part of legit response in the single skb. Now you just drop the response and have potential leak of splitted skb. I think this case is possible if client sent the request that we forwarded and then health monitor timer sent the request. Am I right?

Reply:
It seems that if we can't parse health monitor response, we can't process second response if it is present in the same skb. But it seems we really have a memory leak if spliiting occurs. Also it seems that we can return T_BAD here. not T_OK

Reply:
I fix memory leak here https://github.com/tempesta-tech/tempesta/pull/2212

Reply:
LGTM, but I think we should return T_BAD not T_OK. @const-t how do you think?